Xamarin.Forms.Platform.Android/AppCompat/MasterDetailPageRenderer.cs

```diff
@@ -116,7 +116,7 @@ void IVisualElementRenderer.SetElement(VisualElement element)
 				{
 					_detailLayout = new MasterDetailContainer(newElement, false, Context)
 					{
-						TopPadding = statusBarHeight,
+						TopPadding = Element.Parent is NavigationPage ? 0 : statusBarHeight,
```

Comment:
So what if Parent is not a Navigation, but something like TabbedPage, and that was as a parent a Navigation? Get me?
Nav -> Tab > MDP -> we don't like this in Forms, but in this case your fix doesn't work, maybe a recursive method to check if we have a Nav above us.